purview/utils/teiid.py

```diff
@@ -29,29 +29,44 @@ def __init__(self, host, port, username, password):
         # a dict mapping db names to cursors
         self._connections = {}
 
-    def get_connection(self, db_name, force_new=False):
+    def get_connection(self, db_name, force_new=False, retry=None):
         """
         Return an existing psycopg2 connection and establish it if needed.
 
         :param str db_name: the database name to get a connection to
         :kwarg bool force_new: forces a new database connection even if one
         already exists
+        :kwarg int retry: the number of times to retry a failed connection. If this
+        is not set, then the TEIID connection attempt will be repeated until it is successful.
         :return: a connection to TEIID
         :rtype: psycopg2 connection
         """
         if not force_new and db_name in self._connections:
             return self._connections[db_name]
+        if retry is not None and retry < 1:
```

Comment:
Optional: We can change this to `if retry and retry < 1:`

Reply:
The reason I use `if retry is not None` is because I want this exception to be raised if `retry` is set to `0`.